### PR TITLE
Vault integration: add repository create credential library method

### PR DESCRIPTION
**Pull request by mgaffney**
This also adds the `http_method` and `http_request_body` attributes to the Vault credential library.

### CODE REVIEW

File: internal/credential/vault/repository_credential_library.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+	if l.VaultPath == "" {
+		return nil, errors.New(errors.InvalidParameter, op, "no vault path")
+	}
+	if l.PublicId != "" {
+		return nil, errors.New(errors.InvalidParameter, op, "public id not empty")
+	}
+	if scopeId == "" {
+		return nil, errors.New(errors.InvalidParameter, op, "no scope id")
+	}
+	l = l.clone()
```
Comment (louisruch) [inline internal/credential/vault/repository_credential_library.go:42]: You are calling clone here and then again on line 58 one of them could be dropped

File: internal/db/schema/migrations/postgres/5/02_vault_credential.up.sql [hunk 2]
```diff
@@ -179,6 +198,22 @@ begin;
...
       constraint vault_path_must_not_be_empty
         check(length(trim(vault_path)) > 0),
+    http_method text not null
+      constraint credential_vault_http_method_enm_fkey
+        references credential_vault_http_method_enm (name)
+        on delete restrict
+        on update cascade,
+    http_request_body text
+      constraint http_request_body_only_allowed_with_post_method
+        check(
```
Comment (louisruch) [inline internal/db/schema/migrations/postgres/5/02_vault_credential.up.sql:208]: neat will make a note of this for future!

**Review by louisruch — APPROVED** [review]
LGTM, one comment re cloning twice

File: internal/credential/vault/repository_credential_library.go (hunk 1 above)
Comment (mgaffney) [reply internal/credential/vault/repository_credential_library.go:42]: We do [this](https://github.com/hashicorp/boundary/blob/93d81157758398255baf9715a38fa0b381d975cb/internal/host/static/repository_host_catalog.go#L37) in [other](https://github.com/hashicorp/boundary/blob/93d81157758398255baf9715a38fa0b381d975cb/internal/host/static/repository_host_catalog.go#L71) places too. If I remember correctly, the first clone prevents changes to the object passed in and the second clone is needed for the `w.Create` method.

**Pull request merged**